Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -975,12 +981,17 @@ private void TransactionFoundInternal(Script script)
                         continue;
                     }
 
-                    // calculate how many accounts to add to keep a buffer of 20 unused addresses
+                    // Calculate how many accounts to add to keep a buffer of 20 unused addresses.
                     int lastUsedAddressIndex = account.GetLastUsedAddress(isChange).Index;
                     int addressesCount = isChange ? account.InternalAddresses.Count() : account.ExternalAddresses.Count();
                     int emptyAddressesCount = addressesCount - lastUsedAddressIndex - 1;
                     int accountsToAdd = UnusedAddressesBuffer - emptyAddressesCount;
                     account.CreateAddresses(this.network, accountsToAdd, isChange);
+
+                    this.LoadKeysLookupLock();
```

Comment:
This is where I disagree with Jeremy

This method is inside a lock lock(ProcessTransaction -> AddTransactionToWallet -> TransactionFoundInternal)

Both the methods have their own lock internally
`LoadKeysLookupLock`
`SaveWallet`

Also if a trx has multiple outputs that belong to the wallet we will call the save and index twice.

I suggest we put this code under 
```
   ProcessTransaction()
   { 
           lock (code)

            if (foundTrx.Any())
            {
                this.LoadKeysLookupLock();
                this.SaveWallet()  
            }
  }
```

What do you think mike?
@bokobza will not be available to comment until Saturday so if its an issue we can wait for him (I am happy to wait for Jeremy or do that as a second PR).

Reply:
@dangershony  What do you think about Jeremy's comment here https://github.com/stratisproject/StratisBitcoinFullNode/commit/61c3b92f12afd3fc2b3904af990f73781a6ede69#commitcomment-24893579

Reply:
See my comment, now I am not sure on the next plan of action do we go down the suggested approach by @bokobza or wait for his comments.

Reply:
Agreed lets merge this PR and do further refactoring in another PR.